### PR TITLE
Add configurable finalization

### PR DESCRIPTION
**Pull request by teddyknox**

### CODE REVIEW

**Comment by teddyknox** [comment]
Will move out of draft when this code merges: https://github.com/Layr-Labs/eigenda/pull/620

File: verify/cert.go [hunk 1]
```diff
@@ -1,20 +1,30 @@
...
+var ErrBatchMetadataHashNotFound = errors.New("BatchMetadataHash not found for BatchId")
+
 // CertVerifier verifies the DA certificate against on-chain EigenDA contracts
 // to ensure disperser returned fields haven't been tampered with
 type CertVerifier struct {
-	manager *binding.ContractEigenDAServiceManagerCaller
+	ethConfirmationDepth uint64
+	manager              *binding.ContractEigenDAServiceManagerCaller
+	finalizedBlockClient *FinalizedBlockClient
+	ethClient            *ethclient.Client
```
Comment (epociask) [inline verify/cert.go:27]: Is this client used anywhere in this struct?

File: server/eigenda_store.go [hunk 2]
```diff
@@ -84,6 +95,31 @@ func (e EigenDAStore) Put(ctx context.Context, value []byte) (comm []byte, err e
...
+	dispersalDuration := time.Since(dispersalStart)
+	remainingTimeout := e.cfg.StatusQueryTimeout - dispersalDuration
+
+	ticker := time.NewTicker(12 * time.Second)
+	defer ticker.Stop()
+	ctx, cancel := context.WithTimeout(context.Background(), remainingTimeout)
+	defer cancel()
+
+	done := false
+	for !done {
```
Comment (teddyknox) [inline server/eigenda_store.go:107]: Waiting here

**Review by epociask — APPROVED** [review]
just one knit - otherwise LGTM

File: verify/cert.go (hunk 1 above)
Comment (teddyknox) [reply verify/cert.go:27]: On line 119 yeah

```
blockHeader, err := cv.ethClient.BlockByNumber(context.Background(), nil)
```

**Pull request merged**